homeassistant/components/pyload/__init__.py

```diff
@@ -1 +1,69 @@
-"""The pyload component."""
+"""The pyLoad integration."""
+
+from __future__ import annotations
+
+from aiohttp import CookieJar
+from pyloadapi.api import PyLoadAPI
+from pyloadapi.exceptions import CannotConnect, InvalidAuth, ParserError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+type PyLoadConfigEntry = ConfigEntry[PyLoadAPI]  # noqa: F821
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: PyLoadConfigEntry) -> bool:
+    """Set up pyLoad from a config entry."""
+
+    host = entry.data[CONF_HOST]
+    port = entry.data[CONF_PORT]
+    protocol = "https" if entry.data[CONF_SSL] else "http"
+    url = f"{protocol}://{host}:{port}/"
+
+    session = async_create_clientsession(
+        hass,
+        verify_ssl=False,
+        cookie_jar=CookieJar(unsafe=True),
+    )
+    pyloadapi = PyLoadAPI(
+        session,
+        api_url=url,
+        username=entry.data[CONF_USERNAME],
+        password=entry.data[CONF_PASSWORD],
+    )
+
+    try:
+        await pyloadapi.login()
+    except CannotConnect as e:
+        raise PlatformNotReady(
+            "Unable to connect and retrieve data from pyLoad API"
+        ) from e
+    except ParserError as e:
+        raise PlatformNotReady("Unable to parse data from pyLoad API") from e
+    except InvalidAuth as e:
+        raise PlatformNotReady(
```

Comment:
```suggestion
        raise ConfigEntryError(
```
until you have a reauth flow

Reply:
👍